### ISSUE TITLE
Syntax checking is light on tests

### ISSUE THREAD

**Issue by akrs** [akrs/YACBL#10]
We need more tests for the syntax checker. Make them.

**Comment by akrs** [comment]
@stephenscottsmith @gastyrd @joaquinloustau @mcoccia 

This is your time to shine! :wink:

**Comment by akrs** [comment]
You can now see examples of tests we should have on the aa/parser branch.

**Comment by rtoal** [comment]
Looks like the scanner tests are yak files but the parser tests are json files. Is that intentional?

**Comment by akrs** [comment]
Correct. I wanted to decouple the tests of the parser from the scanner. The json files are the same format that the scanner with produce.
